lib/narn-lib.js

```diff
@@ -22,45 +39,69 @@ exports.detectYarn = function detectYarn(args) {
 
 exports.getYarnArgs = narnArgs => narnArgs;
 
-exports.getNpmArgs = narnArgs => {
-  const yarnArgs = minimist(narnArgs, { boolean: ["dev", "D"] });
+exports.getNpmArgs = (narnArgs, isPnpm = false) => {
+  const yarnArgs = minimist(narnArgs, {
+    boolean: ["dev", "D"],
+    alias: {
+      D: "dev"
+    }
+  });
   const yarnCommands = yarnArgs._;
   const yarnTarget = yarnCommands.length > 0 ? yarnCommands[0] : "install";
   const yarnSubCommands = yarnCommands.slice(1);
+  const narnSubArgs = narnArgs.slice(1);
+
   let npmTarget;
   let npmArgs;
   let result;
   switch (yarnTarget) {
     case "install":
       npmTarget = "install";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
       break;
     case "add":
       npmTarget = "install";
       const packages = yarnSubCommands.map(transformNarnPackageString);
-      const saveType = yarnArgs.dev || yarnArgs.D ? "--save-dev" : "--save";
+      const saveType = yarnArgs.dev ? "--save-dev" : "--save";
       npmArgs = [saveType, ...packages];
       break;
     case "remove":
       npmTarget = "uninstall";
-      npmArgs = yarnSubCommands;
+      npmArgs = narnSubArgs;
```

Comment:
Same here - yarn flags are not the same as npm/pnpm flags. This violates the primary mission of narn which is to never have to worry about what kind of project you're running the command in.

Reply:
one this is the command and another thing is the flags, I repeat, the flags don't kill the functionality, but you just letting minimist eat all the argument that the user might want to add, just because you didn't put EVERY flag on the minimist function call is crazy

Reply:
https://github.com/joeldenning/narn/pull/13#discussion_r376681334